src/kcas.ml

```diff
@@ -194,35 +238,43 @@ module Op = struct
   type t = CAS : 'a Loc.t * 'a * 'a -> t
 
   let make_cas loc before after = CAS (loc, before, after) [@@inline]
+  let make_cmp loc expected = CAS (loc, expected, expected) [@@inline]
 
   let is_on_loc op loc =
     match op with CAS (loc', _, _) -> Obj.magic loc' == loc
```

Comment:
Not added here, so feel free to punt this. We could just compare loc's id's here, rather than use magic, right?

Reply:
I added `Op.get_id` to the PR.